southpark-search/tests/distributed/terraform.tf

```diff
@@ -0,0 +1,204 @@
+provider "aws" {
+  version = "~> 2.0"
+  region  = "us-east-2"
+}
+
+#Create repo
+resource "aws_ecr_repository" "southpark" {
+  name = "sp-repo"
+  tags = {
+    Name = "southpark_repo"
+  }
+}
+
+resource "aws_default_vpc" "default_vpc" {
+}
+
+resource "aws_instance" "encoder" {
+  ami           = "ami-07efac79022b86107"
+  instance_type = "t2.micro"
+  tags = {
+    Name = "encoder"
+  }
+}
+
+resource "aws_instance" "indexer" {
+  ami           = "ami-07efac79022b86107"
+  instance_type = "t2.micro"
+  tags = {
+    Name = "indexer"
+  }
+}
+
+resource "aws_instance" "flow" {
+  ami           = "ami-07efac79022b86107"
+  instance_type = "t2.micro"
+  tags = {
+    Name = "flow"
+  }
+}
```

Comment:
If you add them as plain ec2 instance, the FARGATE ressources are not needed anymore and the initial flow setup can be done via a triggered lambda function or from localhost.

Is `jinad` on these instance reachable from the VPN there via `http://encoder:8000/...`? That would be quite cool.

Reply:
So at the beginning I was using Fargate because it's better for a test env, but I think you are totally right Rutuja, we should try with EC2 instances now.  We won't need the fargate resources but we will need to do the management by ourselves right?

Reply:
Yes we will have to `SSH` in the resources explicitly and run the flow script for spinning up jinaD.
An alternative is to also have a `null resource` in Terraform that can run the script for flow on individual instances.

Reply:
Could you add this null ressource to showcase its usage? that'll be great.

Reply:
Added `local_exec` provisioner to the `flow` instance, `null_resource` could be used for defining specific runs of scripts on different instances, since we have a dedicated instance for `flow` already, I added a `local exec` provisioner.